rust/blockstore/src/arrow/block/delta/delta.rs

```diff
@@ -80,13 +80,13 @@ impl BlockDelta {
     /// The new block deltas contains all the key value pairs after, but not including the
     /// split point.
     pub(crate) fn split<K: ArrowWriteableKey, V: ArrowWriteableValue>(
-        &self,
+        self,
```

Comment:
consuming `self` here makes sense to me as a way to signal that the delta is consumed, please let me know if this is not semantically correct (similarly, why are deltas clonable?)